File: src/main/java/pl/yalgrin/playnite/simplesync/repository/SourceRepository.java
package pl.yalgrin.playnite.simplesync.repository;

import org.springframework.data.r2dbc.repository.Query;
import org.springframework.stereotype.Repository;
import pl.yalgrin.playnite.simplesync.domain.Source;
import reactor.core.publisher.Flux;

@Repository
public interface SourceRepository extends ObjectRepository<Source> {
    @Query("select c.id from playnite_source c order by c.id asc")
    Flux<Long> findAllIds();
}

File: src/main/kotlin/pl/yalgrin/playnite/simplesync/library/repository/SourceRepository.kt
package pl.yalgrin.playnite.simplesync.library.repository

import org.springframework.data.r2dbc.repository.Query
import org.springframework.stereotype.Repository
import pl.yalgrin.playnite.simplesync.library.domain.Source
import reactor.core.publisher.Flux

@Repository
interface SourceRepository : ObjectRepository<Source> {
    @Query("select c.id from playnite_source c order by c.id")
    override fun findAllIds(): Flux<Long>
}
